lib/mlir/Dialect/Clift/IR/ImportModel.cpp

```
@@ -0,0 +1,599 @@
+
+  ~CliftConverter() {
+    revng_assert(DefinitionGuardSet.empty());
+    revng_assert(IncompleteTypes.empty());
+  }
+
+  clift::ValueType convertType(const model::Type &ModelType) {
+    const clift::ValueType T = getUnwrappedType(ModelType,
+                                                /*RequireComplete=*/true);
+    if (T and not processIncompleteTypes()) {
+      IncompleteTypes.clear();
+      return nullptr;
+    }
+    return T;
+  }
+
+  clift::ValueType convertQualifiedType(const model::QualifiedType &ModelType) {
+    const clift::ValueType T = getQualifiedType(ModelType,
+                                                /*RequireComplete=*/true);
+    if (T and not processIncompleteTypes()) {
+      IncompleteTypes.clear();
+      return nullptr;
+    }
+    return T;
+  }
+
+private:
+  mlir::BoolAttr getBool(bool const Value) {
+    return mlir::BoolAttr::get(Context, Value);
+  }
+
+  mlir::BoolAttr getFalse() { return getBool(false); }
+
+  template<typename T, typename... ArgTypes>
+  T make(const ArgTypes &...Args) {
+    if (EmitError and failed(T::verify(EmitError, Args...)))
+      return {};
+    return T::get(Context, Args...);
+  }
+
+  static clift::PrimitiveKind
+  getPrimitiveKind(const model::PrimitiveTypeKind::Values K) {
+    switch (K) {
+    case model::PrimitiveTypeKind::Void:
+      return clift::PrimitiveKind::VoidKind;
+    case model::PrimitiveTypeKind::Generic:
+      return clift::PrimitiveKind::GenericKind;
+    case model::PrimitiveTypeKind::PointerOrNumber:
+      return clift::PrimitiveKind::PointerOrNumberKind;
+    case model::PrimitiveTypeKind::Number:
+      return clift::PrimitiveKind::NumberKind;
+    case model::PrimitiveTypeKind::Unsigned:
+      return clift::PrimitiveKind::UnsignedKind;
+    case model::PrimitiveTypeKind::Signed:
+      return clift::PrimitiveKind::SignedKind;
+    case model::PrimitiveTypeKind::Float:
+      return clift::PrimitiveKind::FloatKind;
+    default:
+      revng_abort();
```

`revng_abort` can also take a string literal used as error message. Add something meaningful here.

---

I fixed this and added an aborting case for the `Invalid` and `Count` fields instead of using the default case. This way if any new fields are added in the future we get an error here.